gcsfs/core.py

```diff
@@ -537,25 +538,32 @@ async def _get_object(self, path):
                 raise FileNotFoundError(path)
         return self._process_object(bucket, res)
 
-    async def _list_objects(self, path, prefix="", versions=False):
+    async def _list_objects(self, path, prefix="", versions=False, **kwargs):
+
         bucket, key, generation = self.split_path(path)
         path = path.rstrip("/")
 
-        try:
-            clisting = self._ls_from_cache(path)
-            hassubdirs = clisting and any(
-                c["name"].rstrip("/") == path and c["type"] == "directory"
-                for c in clisting
-            )
-            if clisting and not hassubdirs:
-                return clisting
-        except FileNotFoundError:
-            # not finding a bucket in list of "my" buckets is OK
-            if key:
-                raise
+        # NOTE: the inventory report logic is experimental.
+        inventory_report_info = kwargs.get("inventory_report_info", None)
+
+        # Only attempt to list from the cache when the user does not use
```

Comment:
Yes, the inventory report listing results (if `use_snapshot_listing` is set to `False`, and thus it actually uses coroutines to fetch most up-to-date result) is cached, refer to the end of `list_objects` method.

When `use_snapshot_listing` is set to `True`, it is slightly complicated. Since the inventory reports might not contain all, but only potentially **a subset of** the object metadata (the user decides what object metadata they want to include the inventory report when they set up the inventory report in the console), I've decided not to add it to the cache, since that subset does not fully represent everything in the directory.

But maybe what we could do is instead not not looking up the cache whenever the inventory report option is configured, we only use the cache when `use_snapshot_listing` is set to `False`. 

So instead of the `if not inventory_report_info:` check, we do `if not inventory_report_info or not inventory_report_info["use_snapshot_listing"]. 

Thoughts?

Reply:
Leave it for now. I think there may be a use case for caching the manifest inventory and relying on it entirely, but we can see if anyone asks for that.